pkg/machines/components/vm/disks/diskAdd.jsx

```diff
@@ -350,7 +350,7 @@ export class AddDiskModalBody extends React.Component {
         const stateDelta = { existingVolumeName: value };
         const pool = storagePools.find(pool => pool.name === poolName && pool.connectionName === vm.connectionName);
         stateDelta.format = getDefaultVolumeFormat(pool);
-        if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
+        if (pool && ['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
```

Comment:
getDefaultVolumeFormat also accesses pool.type so this won't fix the crash. You should probably do:
```
if (!pool)
    return;
````